modules/nixops4Resource/nixos.nix

```diff
@@ -41,6 +41,15 @@ in
       type = types.str;
       default = "";
     };
+    sudo.enable = mkOption {
+      type = types.bool;
+      default = config.ssh.user != "root";
+      defaultText = lib.literalMD ''`false` iff `config.ssh.user` is `"root"`'';
+      description = ''
+        Whether to call `sudo` when applying the configuration.
+        This is necessary when the user is not `root`.
```

Comment:
Isn't that saying that `sudo.enable` should always be set to `ssh.user != "root"`? Then why would it be an option?

Reply:
I guess it's too much info really. It's almost common sense, something that NixOS should document, and the default value says enough.

```suggestion
```

> Then why would it be an option

I think we'll end up making `ssh.user = null;` a thing, so that local SSH config can decide the user name, and I feel like in that case we'll want to provide this option.
It is perhaps slightly over-engineered until we have that, but I think it's ok.

Reply:
> I think we'll end up making `ssh.user = null;` a thing, so that local SSH config can decide the user name, and I feel like in that case we'll want to provide this option.
> It is perhaps slightly over-engineered until we have that, but I think it's ok.

Yeah, I don't think it's bad. I suppose with `ssh.user = null;` we'd just take the local user or whatever comes from the local SSH config; but then there is a chance that the local SSH config could specific a `root` or non-`root` user, and we'd probably still want to use `sudo` in the latter case but not in the former? I'd almost want something like `sudo.enable = false | true | auto` where `auto` decides after we know what actual user is being used.

Reply:
Sounds fair, it can always be changed/refined later. `ssh.user = null;` sounds like what we want.